tests/phpt/smart_casts/05_unions.php

```diff
@@ -0,0 +1,123 @@
+@ok
+<?php
+
+class Expr {}
+
+class UnaryOp extends Expr {
+  public $arg = null;
+
+  public function __construct(Expr $arg) {
+    $this->arg = $arg;
+  }
+}
+class UnaryMinus extends UnaryOp {}
+class UnaryPlus extends UnaryOp {}
+
+class Literal extends Expr {}
+class StringLiteral extends Literal { public $string_value = 'foo'; }
+class IntLiteral extends Literal { public $int_value = 54; }
+
+/** @param StringLiteral|IntLiteral $x */
+function test1($x) {
+  if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);
+  }
+}
+
+/** @param StringLiteral|IntLiteral|UnaryOp $x */
+function test2($x) {
+  if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);
+  } else if ($x instanceof UnaryOp) {
+    var_dump(get_class($x->arg));
+  }
+}
+
+/** @param StringLiteral|UnaryOp|IntLiteral $x */
+function test3($x) {
+  if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);
+  } else if ($x instanceof UnaryOp) {
+    var_dump(get_class($x->arg));
+  }
+}
+
+/** @param UnaryOp|Literal $x */
+function test4($x) {
+  if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);
+  } else if ($x instanceof UnaryOp) {
+    var_dump(get_class($x->arg));
+  }
+}
+
+/** @param Literal|UnaryOp $x */
+function test5($x) {
+  if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);
+  } else if ($x instanceof UnaryOp) {
+    var_dump(get_class($x->arg));
+  }
+}
+
+/** @param UnaryMinus|Literal|UnaryPlus $x */
+function test6($x) {
+  if ($x instanceof UnaryPlus) {
+    var_dump(['unary plus' => get_class($x->arg)]);
+  } else if ($x instanceof StringLiteral) {
+    var_dump($x->string_value);
+  } else if ($x instanceof IntLiteral) {
+    var_dump($x->int_value);} else if ($x instanceof UnaryMinus) {
```

Comment:
formatting

Reply:
Fixed. Thanks for noticing. It's actually not only a formatting issue, it was a redundant branch that made another one unreachable.